include/rapidjson/document.h

```diff
@@ -952,12 +956,16 @@ class GenericValue {
         if (IsUint64()) {
             uint64_t u = GetUint64();
             volatile double d = static_cast<double>(u);
-            return static_cast<uint64_t>(d) == u;
+            return (d >= 0.0)
+                && (d < (static_cast<double>(std::numeric_limits<uint64_t>::max()/2 + 1)) * 2)
```

Comment:
I am not sure about the calculation `(max / 2 + 1) * 2`. I think it can simply construct a maximum double that can be converted to `uint64_t`.